### PR TITLE
Xen port improvements made in xen18 branch

### PR DESCRIPTION
**Pull request by AntonKozlov**

### CODE REVIEW

File: src/arch/xen/include/COPYING [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Source files taken from the Xen distribution are governed by their original license.
```
Comment (anton-bondarev) [inline src/arch/xen/include/COPYING]: This is unusual file or place for it

File: src/arch/xen/include/barrier.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/*
```
Comment (anton-bondarev) [inline src/arch/xen/include/barrier.h]: include guard?

File: src/arch/xen/include/xenstore.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#include <stdint.h>
```
Comment (anton-bondarev) [inline src/arch/xen/include/xenstore.h]: include guard?

File: src/arch/xen/include/COPYING (hunk 1 above)
Comment (AntonKozlov) [reply src/arch/xen/include/COPYING]: It belong to Xen port at least. I have no strong opinion where would it belong. `src/arch/xen` (one level upper)?

Comment (anton-bondarev) [reply src/arch/xen/include/COPYING]: Yes, one level upper looks better.
But it means than all code in xen folder is under this license?

File: src/arch/xen/include/xenstore.h (hunk 3 above)
Comment (AntonKozlov) [reply src/arch/xen/include/xenstore.h]: fixed as well, thanks!

File: src/arch/xen/include/barrier.h (hunk 2 above)
Comment (AntonKozlov) [reply src/arch/xen/include/barrier.h]: thanks, fixed

File: src/arch/xen/include/COPYING (hunk 1 above)
Comment (AntonKozlov) [reply src/arch/xen/include/COPYING]: License specifies xenstore related files, it's xensotre.c and xenstore.h

**Comment by AntonKozlov** [comment]
Just to make sure, COPYING moved to src/arch/xen/ is better, but is it good enough? Please duplicate LGTM after the move :)

**Comment by anton-bondarev** [comment]
duplicate LGTM :)

**Pull request merged**